java/vector/src/main/java/org/apache/arrow/vector/NullableTimestampVector.java

```diff
@@ -179,6 +258,37 @@ public static long get(final ArrowBuf buffer, final int index) {
     return buffer.getLong(index * TYPE_WIDTH);
   }
 
+  public void get(int index, NullableTimestampHolder holder) {
+    if (isSet(index) == 0) {
+      holder.isSet = 0;
+      return;
+    }
+    holder.isSet = 1;
+    holder.value = valueBuffer.getLong(index * TYPE_WIDTH);
+  }
+
+  @Override
+  public LocalDateTime getObject(int index) {
+    if (isSet(index) == 0) {
+      return null;
+    } else {
+      long millis = unit.getTimeUnit().toMillis(get(index));
```

Comment:
Would it be more efficient to have a `toMillis` method on `TimeUnit` (and encapsulate this detail per law of demeter)? I guess we need to run some microbenchmarks to be able to judge performance

Reply:
Good point. I encapsulated  java TimeUnit detail.

Also I ran a microbench toMilliis for 1B long values. Performance is the same.